examples/feature-factory/tsconfig.json

```diff
@@ -1,12 +1,19 @@
 {
   "compilerOptions": {
-    "target": "ESNext",
-    "module": "ESNext",
-    "moduleResolution": "node",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "lib": ["ES2022"],
     "resolveJsonModule": true,
     "esModuleInterop": true,
+    "isolatedModules": true,
     "strict": true,
-    "rootDir": "."
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitOverride": true,
+    "exactOptionalPropertyTypes": true,
+    "noUncheckedIndexedAccess": true,
+    "noFallthroughCasesInSwitch": true,
+    "skipLibCheck": true
   },
```

Comment:
Might be nice to break these out into sections and add labels/comments, similar to how our internal tsconfigs look.